### ISSUE TITLE
Align `Line_Ending_Style`

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4465]
This task is automatically imported from the old Task Issue Board and it was originally created by Radosław Waśko.
Original issue is [here](https://www.pivotaltracker.com/story/show/182749831).
<hr />

1. By mistake I added one more instance of `Line_Ending_Style` but it already existed in the stdlib - the new one has to be removed and the usages need to be aligned with the existing one.

2. As discussed, we should change the default line endings to be a constant (Unix) instead of platform dependent. The change should not affect read mode where the default should still be to infer the endings from the file contents. So we will keep `Infer` as the default value, but in write mode it will default to Unix endings instead of platform specific. Tests need to be updated accordingly.

## Comments:
**Radosław Waśko** reports a new **STANDUP** for the provided date (2022-07-19):

**Progress:** Addressing PR reviews for pgpass, aligned line endings after our discussion (another PR created for that, just waiting in queue to be merged), started some work on Index_Sub_Range refactor It should be finished by 2022-07-21.

**Next Day:** Next day I will be working on the [##182306743 task](https://www.pivotaltracker.com/story/show/182306743). Make sure the pending PRs get merged, continue work on Index_Sub_Range (Enso Bot - Jul 21, 2022)
<hr />
**Radosław Waśko** reports a new **🔴 DELAY** for yesterday (2022-07-21):

**Summary:** There is **1 day** delay in implementation of the [Align `Line_Ending_Style` (#182749831) task](https://www.pivotaltracker.com/story/show/182749831).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Turns out that while the logic worked on Linux, there is a edge case on Windows where the line endings were wrongly detected. It was a pretty complicated one to detect properly in a reasonable way - some additional work is needed to handle it properly. (Enso Bot - Jul 22, 2022)
<hr />
**Radosław Waśko** reports a new **STANDUP** for yesterday (2022-07-21):

**Progress:** Got back to the line endings issue, as it turns out there is a pretty hard to tackle edge case on Windows where the line endings were detected wrongly. Analysed the situation and tried implementing a solution that should fix it. As this is very highly related, also tackled #182780570. Added tests, but still some fine tuning is needed to make them pass. It should be finished by 2022-07-22.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/182749831). Fix the remaining issues, verify that the logic for line ending detection is correct and add tests for problematic edge cases if any additional ones are found. Update the PR and get it merged. (Enso Bot - Jul 22, 2022)
<hr />
**Radosław Waśko** reports a new **🔴 DELAY** for today (2022-07-25):

**Summary:** There is **4 days** delay in implementation of the [Align `Line_Ending_Style` (#182749831) task](https://www.pivotaltracker.com/story/show/182749831).
It **will cause 1 day delay** for the delivery of this weekly plan.

**Delay Cause:** The parser was uncooperating so it took me a bit more time than I'd like to find a viable workaround; then I also got stuck on some polyglot exception bug which hindered debugging a lot. Now, the PR is reviewed but I need to implement a more efficient byte array representation and so I need to discuss this with Jaroslav - so will prepare for a merge tomorrow. (Enso Bot - Jul 25, 2022)
<hr />
**Radosław Waśko** reports a new **STANDUP** for the provided date (2022-07-22):

**Progress:** Added more edge cases, was fighting with the parser not detecting line endings how we want, finally found an acceptable solution but got a bit stuck on some weird issue where I was catching an exception "Polyglot object" instead of an actual exception (which turned out to be an NPE) which made it pretty hard to debug. It should be finished by 2022-07-26.

**Next Day:** Next day I will be working on the [#same task](https://www.pivotaltracker.com/story/show/182749831). Figure out the error, fix it. Try to find a repro for the exception bug. Prepare the line endings PR for another review. (Enso Bot - Jul 25, 2022)
<hr />
**Radosław Waśko** reports a new **STANDUP** for today (2022-07-25):

**Progress:** Finally fixed the edge cases and got the PR reviewed. Applied some suggestions, rewritten the array conversion to ArrayOverBuffer as Jaroslav suggested. Created a repro for the exception issue that was hindering my debugging. It should be finished by 2022-07-26.

**Next Day:** Next day I will be working on the [##182306743 task](https://www.pivotaltracker.com/story/show/182306743). Discuss the ArrayOverBuffer solution to make sure it's ok and get the PR merged. Go back to working on Index_Sub_Range. (Enso Bot - Jul 25, 2022)
<hr />
**Radosław Waśko** reports a new **STANDUP** for today (2022-07-26):

**Progress:** Started design work on the new sampling methods (Index_Sub_Range), discussed the specification in more detail and drafted an approach that should work well. In the meantime it turned out the line endings were still a problem - the tests were failing on Windows still (I should have double checked that, I forgot, sorry). So I switched back to finishing this PR, I think I finally managed to find a pretty elegant solution for checking if we can 'trust' the detected newline from the parser. I added a few more edge case tests just to be sure the behaviour is correct and verified on Linux (using a trick to emulate the situation encountered on Windows) that the new solution should work. Now just waiting on CI to verify that it indeed works on Windows and to merge the PR finally... It should be finished by 2022-07-26.

**Next Day:** Next day I will be working on the [##182306743 task](https://www.pivotaltracker.com/story/show/182306743). Create an initial shape for the Index_Sub_Range implementations for Text.take and Text.drop and discuss it with James, continue with the implementation and adding tests for the new functionalities. (Enso Bot - Jul 26, 2022)
<hr />
